### ISSUE TITLE
Clean up tests on PyPy

### ISSUE THREAD

**Issue by aaugustin** [python-websockets/websockets#1483]
They're failing too often in CI.

Also there's spurious warnings.

**Comment by aaugustin** [comment]
Frequent warning:

```
~/.pyenv/versions/pypy3.10-7.3.16/lib/pypy3.10/asyncio/sslproto.py:320: ResourceWarning: unclosed transport <asyncio.sslproto._SSLProtocolTransport object at 0x...>
```

Test failure not linked to a specific test, but to the framework:

```
Traceback (most recent call last):
  File "/Users/myk/dev/websockets/tests/legacy/test_client_server.py", line 122, in _decorate
    return func(self, *_args, **_kwargs)
  File "/Users/myk/dev/websockets/tests/legacy/test_client_server.py", line 121, in _decorate
    with manager(self, *args, **kwargs):
  File "/Users/myk/.pyenv/versions/pypy3.10-7.3.16/lib/pypy3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/Users/myk/dev/websockets/tests/legacy/test_client_server.py", line 105, in temp_test_client
    test.start_client(*args, **kwargs)
  File "/Users/myk/dev/websockets/tests/legacy/test_client_server.py", line 331, in start_client
    super().start_client(path, **kwargs)
  File "/Users/myk/dev/websockets/tests/legacy/test_client_server.py", line 266, in start_client
    self.assertDeprecationWarnings(recorded_warnings, expected_warnings)
  File "/Users/myk/dev/websockets/tests/legacy/utils.py", line 80, in assertDeprecationWarnings
    self.assertEqual(type(recorded.message), DeprecationWarning)
AssertionError: <class 'ResourceWarning'> != <class 'DeprecationWarning'>
```

**Comment by aaugustin** [comment]
I doesn't make sense to spend much time fixing the test suite for the legacy implementation under PyPy, though.

**Comment by aaugustin** [comment]
This warning is a bug: https://github.com/python/cpython/issues/90476

It was fixed in Python 3.11.

I tried to remove it by adding this to `tests/__init__.py`:

```python
import warnings
warnings.filterwarnings(action="ignore", category=ResourceWarning)
```

This filters out ResourceWarning. I could confirm by raising a `ResourceWarning` just after this line. However, the warnings still get raised later in the tests. Probably something changes the warning filters inbetween. I just don't know what.

Removing warnings with command line arguments (`pypy3.9 -W ignore:"unclosed transport":ResourceWarning:asyncio.sslproto -m unittest`) works but that's quite annoying and hard to make conditional on the Python version. Very messy.
